rust/arrow/test/data/various_types.csv

```diff
@@ -1,6 +1,6 @@
-c_int|c_float|c_string|c_bool
-1|1.1|"1.11"|true
-2|2.2|"2.22"|true
-3||"3.33"|true
-4|4.4||false
-5|6.6|""|false
\ No newline at end of file
+c_int|c_float|c_string|c_bool|c_date
```

Comment:
I suggest we also add a c_datetime column too to cover the schema inference of that type too

Reply:
Added, added some extra tests / checks as well.